singer-io/tap-tiktok-ads
# TDL-17938 Updated bookmarking strategy

# Description of change
- Updated the bookmarking strategy for advertisers, ads, adgroups and campaigns as it was returning None when no state was passed
- Updated the comparison strategy in ads, adgroups and campaings
- Added unittests

# Manual QA steps
 - Check that the tap works in case of no state is passed
 - Check that the streams ads, adgroups and campaings replicate properly when bookmark is passed.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
